### DISCUSSION TITLE
How can I preprocess block data to use in vue components in website frontend (and preview in backend)

### DISCUSSION THREAD

**Discussion by dweerd** [area17/twill#1430] [Q&A] [answered]
I just started using Twill, and made an FAQ block, with a repeater for questions and answers. In the frontend of the website I was trying to display this using a custom <faq> vue component that I put in the block blade template, but I am struggling with how to feed the block items into this component.

I could put something like this in `resources/views/site/blocks/faq.blade.php`:
`<faq items="{{ $block->children }}" />`

But that would put in a lot of data that the faq component doesn't need (blockable_id, child_key, editor_name etc). I just want to pass an array of questions and answers. Is there any place where I can do a bit of preprocessing before the block gets rendered? Of course, this should work for the preview in the admin as well.

Thanks!

**Comment by ifox** [comment, marked as answer]
Hi @dweerd, that's something you can do in the block view directly using the @php blade directive, or soon using a PHP class thanks to #1421.

**Comment by dweerd** [comment]
Thanks, that was what I was looking for!
